src/store.js

```diff
@@ -1,16 +1,22 @@
 // We create Redux store directly, instead of using any extra dependencies.
-import { legacy_createStore as createStore, applyMiddleware, compose } from 'redux';
+import { applyMiddleware, compose, legacy_createStore as createStore } from 'redux';
 import thunk from 'redux-thunk';
 import createReducer from './reducers';
 import * as analytics from './analytics/analytics';
 import appSettings from './config/settings';
+import * as uppyReduxStore from '@uppy/store-redux';
 
+let storeInstance = null;
 /**
  * Create a new store with the given initial state. Adds Redux
  * middleware and enhancers.
  */
 export default function configureStore(initialState = {}, sdk = null, analyticsHandlers = []) {
-  const middlewares = [thunk.withExtraArgument(sdk), analytics.createMiddleware(analyticsHandlers)];
+  const middlewares = [
+    thunk.withExtraArgument(sdk),
+    analytics.createMiddleware(analyticsHandlers),
+    uppyReduxStore.middleware(),
```

Comment:
Uppy is connected with the main Redux store, so we have only one source of truth, as it's suggested by Redux guidelines.